build.sc

```diff
@@ -844,9 +844,23 @@ trait CliIntegration extends SbtModule with ScalaCliPublishModule with HasTests
       def test(args: String*) = {
         val argsTask = T.task {
           val launcher = launcherTask().path
+          val debugReg = "--debug(:[0-9]+)?".r
+          val debugOpt = args.filter(a => debugReg.pattern.matcher(a).matches).headOption
+          val (isDebug, debugPort) = debugOpt match {
+            case Some(opt: String) => opt.split(":", 2) match {
```

Comment:
NIT Better not type things in match cases, unless it's a type check (that is, the object might be something of an other instance sometimes, and we check that it has the right type at runtime). If ever the type of `debugOpt` unexpectedly becomes `Option[Any]`, we may not detect it, as the pattern makes it look as if it contains a string (while it may not anymore).

If you really want to write some explicit types here, these can be added to some of the `val`s (like `val debugOpt`, but not `val (isDebug, debugPort)`, as it's a pattern too).